src/cryptography/hazmat/bindings/openssl/binding.py

```diff
@@ -74,6 +87,7 @@ def _register_osrandom_engine(cls):
     def _ensure_ffi_initialized(cls):
         with cls._init_lock:
             if not cls._lib_loaded:
+                cls.lib = _ConditionalLibrary()
```

Comment:
I know this will work as it's written, but I think it's cleaner to pass the lib and ``CONDITIONAL_NAMES`` dictionary into the ``_ConditionalLibrary``. Easier to test as well!